compiler/rustc_query_system/src/ich/impls_hir.rs

```diff
@@ -47,28 +46,6 @@ impl<'ctx> rustc_hir::HashStableContext for StableHashingContext<'ctx> {
         })
     }
 
-    #[inline]
-    fn hash_hir_mod(&mut self, module: &hir::Mod<'_>, hasher: &mut StableHasher) {
-        let hcx = self;
-        let hir::Mod { inner: ref inner_span, ref item_ids } = *module;
-
-        inner_span.hash_stable(hcx, hasher);
-
-        // Combining the `DefPathHash`s directly is faster than feeding them
-        // into the hasher. Because we use a commutative combine, we also don't
-        // have to sort the array.
-        let item_ids_hash = item_ids
-            .iter()
-            .map(|id| {
-                let def_path_hash = id.to_stable_hash_key(hcx);
-                def_path_hash.0
-            })
-            .fold(Fingerprint::ZERO, |a, b| a.combine_commutative(b));
```

Comment:
`combine_commutative`can be removed, since it isn't used anywhere else.

Reply:
I thought that it might be used again in the future, so I kept it. cc @michaelwoerister

Reply:
Yes, let's keep it please. I might have a concrete use for it soon.

Reply:
For similar use case that ignores order of combined fingerprints? In that case we should fix the implementation instead of leaving it broken: when used as here it doesn't ignore the order when combining more than two fingerprints.

Reply:
Opened to #92528 to remove unintentional swap of lower and upper part of the result.